### ISSUE TITLE
Handle SIGTERM / SIGINT signal gracefully

### ISSUE THREAD

**Issue by setraj** [arvyshka/blazer#47]
The process doesn't clean up the temp directory created for segment storage when it receives SIGTERM / SIGINT signal.

<img width="433" alt="Screenshot 2021-08-10 at 7 35 18 PM" src="https://user-images.githubusercontent.com/6270957/128882061-c93569d2-fd99-4078-a6be-f753ebd7f9ac.png">

**Comment by arvyshka** [comment]
I didn't handle this case because, i wanted to give user chance to continue from where it is left off, if he re-runs the command, maybe we can leave a note to user. 

Sometimes people accidently press (control + c) or even if our program panics and cancels, why delete the files so the user has to spend time again to start from beginning.

**Comment by setraj** [comment]
Once the process is interrupted it is going to leave segment files (potentially uncompleted) that are not usable in the next try if I understand correctly?

**Comment by arvyshka** [comment]
> Once the process is interrupted it is going to leave segment files (potentially uncompleted) that are not usable in the next try if I understand correctly?

Good point, we need to test this doubt and also read necessary docs for io library. Please create an issue for this we will investigate. but if there is a log "Downloaded segment ..",  that proves segment downloaded successfully

![image](https://user-images.githubusercontent.com/4346153/128919994-2fb593e3-6254-4846-a0e2-693cb39466e2.png)



> because next time we create a new session.

Actually no, session ID = Hash(URL,threadcount) . so, with same params, we get same session ID and hence download resumes, we infact show a message to user that we are resuming download
